ui/preferences/src/main/java/de/danoeh/antennapod/ui/preferences/screen/bugreport/BugReportFragment.java

```diff
@@ -0,0 +1,227 @@
+
+                if (state == BugReportViewModel.UiState.CrashLogState.SHOWN_COLLAPSED) {
+                    viewBinding.expandCrashLogButton.setText(R.string.report_bug_crash_log_expand);
+                    viewBinding.crashLogContentText.setMaxLines(4);
+                } else {
+                    viewBinding.expandCrashLogButton.setText(R.string.report_bug_crash_log_collapse);
+                    viewBinding.crashLogContentText.setMaxLines(Integer.MAX_VALUE);
+                }
+
+                viewBinding.crashLogToggleGroup.setVisibility(View.VISIBLE);
+                break;
+
+            default:    // UNAVAILABLE
+                viewBinding.crashLogToggleGroup.setVisibility(View.GONE);
+                break;
+        }
+    }
+
+    private void setupContextMenu() {
+        requireActivity().addMenuProvider(new MenuProvider() {
+            @Override
+            public void onCreateMenu(@NonNull Menu menu, @NonNull MenuInflater menuInflater) {
+                menuInflater.inflate(R.menu.bug_report_options, menu);
+            }
+
+            @Override
+            public boolean onMenuItemSelected(@NonNull MenuItem menuItem) {
+                if (menuItem.getItemId() == R.id.export_logcat) {
+                    MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(requireContext());
+
+                    builder.setTitle(menuItem.getTitle());
+                    builder.setMessage(R.string.confirm_export_log_dialog_message);
+                    builder.setPositiveButton(R.string.confirm_label, (dialog, which) -> {
+                        exportLogcat();
+                        dialog.dismiss();
+                    });
+
+                    builder.setNegativeButton(R.string.cancel_label, null);
+                    builder.show();
+
+                    return true;
+                }
+
+                return false;
+            }
+        }, getViewLifecycleOwner());
+    }
+
+    private void setupClipboardCopy() {
+        Context ctx = requireContext();
+
+        viewBinding.attribAppVersionLabel.setOnClickListener(
+                new ClipboardUtils.TextViewCopyOnClickListener(ctx, R.string.report_bug_attrib_app_version));
+        viewBinding.attribAndroidVersionLabel.setOnClickListener(
+                new ClipboardUtils.TextViewCopyOnClickListener(ctx, R.string.report_bug_attrib_android_version));
+        viewBinding.attribDeviceNameLabel.setOnClickListener(
+                new ClipboardUtils.TextViewCopyOnClickListener(ctx, R.string.report_bug_attrib_device_name));
+
+        viewBinding.crashLogContentText.setOnClickListener(new ClipboardUtils.ViewCopyOnClickListener(ctx,
+                R.string.report_bug_crash_log_title, R.string.copied_to_clipboard) {
+            @Override
+            protected String getText(View view) {
+                return viewModel.requireCurrentState().getCrashInfoWithMarkup();
+            }
+        });
```

Comment:
This seems a bit complex. How about extracting a new method `ClipboardUtils.copy(text, title)` and simply calling this using a normal OnClickListener? Then we can also use a  lambda here.